homeassistant/components/nut/config_flow.py

```diff
@@ -176,17 +170,66 @@ def _host_port_alias_already_configured(self, user_input: dict[str, Any]) -> boo
 
     async def _async_validate_or_error(
         self, config: dict[str, Any]
-    ) -> tuple[dict[str, Any], dict[str, str]]:
-        errors = {}
-        info = {}
+    ) -> tuple[dict[str, Any], dict[str, str], dict[str, str]]:
+        errors: dict[str, str] = {}
+        info: dict[str, Any] = {}
+        description_placeholders: dict[str, str] = {}
         try:
             info = await validate_input(self.hass, config)
-        except CannotConnect:
+        except NUTLoginError:
+            errors[CONF_PASSWORD] = "invalid_auth"
+        except NUTError as ex:
             errors[CONF_BASE] = "cannot_connect"
+            description_placeholders["error"] = str(ex)
+        except AbortFlow:
+            raise
         except Exception:  # pylint: disable=broad-except
             _LOGGER.exception("Unexpected exception")
             errors[CONF_BASE] = "unknown"
-        return info, errors
+        return info, errors, description_placeholders
+
+    async def async_step_reauth(
+        self, entry_data: Mapping[str, Any]
+    ) -> ConfigFlowResult:
+        """Handle reauth."""
+        entry_id = self.context["entry_id"]
+        self.reauth_entry = self.hass.config_entries.async_get_entry(entry_id)
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle reauth input."""
+        errors: dict[str, str] = {}
+        existing_entry = self.reauth_entry
+        assert existing_entry
+        existing_data = existing_entry.data
+        description_placeholders: dict[str, str] = {
+            CONF_HOST: existing_data[CONF_HOST],
+            CONF_PORT: existing_data[CONF_PORT],
+        }
+        if user_input is not None:
+            new_config = {
+                **existing_data,
+                # Username/password are optional and some servers
+                # user ip based authentication and will fail if
+                # username/password are provided
```

Comment:
I think "user" is supposed to be "use" here(?)
```suggestion
                # Username/password are optional and some servers
                # use ip based authentication and will fail if
                # username/password are provided
```

Reply:
thanks